app/models/order.rb

```diff
@@ -4,8 +4,15 @@ class Order < ApplicationRecord
   has_many :items, through: :order_items
   has_many :order_items, dependent: :destroy
 
+  def order_total
+    if status == 'open'
+      order_items.map { |oi| oi.item[:price] * oi[:quantity].to_f }.reduce(0.0, :+)
+    else
+      order_items.map { |oi| oi[:purchase_price] * oi[:quantity].to_f }.reduce(0.0, :+)
+    end
+  end
 
-	def order_total
-		self.order_items.map {|oi| oi.item[:price] * oi[:quantity]}.reduce(0,:+)
-	end
+  def handle_payment
+    order_items.each { |oi| oi.purchase_price = oi.item.price; oi.save }
```

Comment:
I believe we can do 1 db insert (instead of n) by moving the `save` out of the block. Give it a try.